### PR TITLE
Add e2e tests for merchant edit order flows

### LINKED ISSUE #27877: Write new e2e test: Merchant / Orders flow (view order, update status, update order details)
Write a new e2e test that covers Merchant / Orders flow. 

The test should cover the following:
- View single order
- Update order status
- Update order details

### PR DESCRIPTION
**Pull request by tammullen**
Replaces https://github.com/woocommerce/woocommerce/pull/28348

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This adds new e2e tests that cover the following Merchant / Orders flows.
View single order
Update order status
Update order details

Closes https://github.com/woocommerce/woocommerce/issues/27877 .

### How to test the changes in this Pull Request:

1. Make sure e2e tests pass locally and on Travis. Run tests as per [our docs](https://github.com/woocommerce/woocommerce/tree/master/tests/e2e#running-tests). 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

N/A

### CODE REVIEW

**Review by rrennick — DISMISSED** [review]
@tammullen Works great. This needs to be updated from `StoreOwnerFlow` to `merchant` before merging.

**Comment by tammullen** [comment]
> @tammullen Works great. This needs to be updated from `StoreOwnerFlow` to `merchant` before merging.

Thanks! Updated.

**Review by zhongruige — COMMENTED** [review]
Thanks for adding these in @tammullen!

I just had a couple of comments, feel free to let me know if you have questions or anything else!

One other thing I did notice was the pipeline has been failing. When I ran the tests locally it passed. For example, the pipeline is failing on the refund step:

```
 WooCommerce Orders > Refund an order

    ✓ can issue a refund by quantity (1713ms)

    ✕ can delete an issued refund (932ms)

  WooCommerce Orders > Apply coupon

    ✓ can apply a coupon (616ms)

    ✓ can remove a coupon (201ms)

  ● WooCommerce Orders > Refund an order › can delete an issued refund

    TimeoutError: Element .quantity .refunded (text: "-1") found

      waiting for function failed: timeout 500ms exceeded
```

However, locally, it ran fine:

```
  WooCommerce Orders > Refund an order
    ✓ can issue a refund by quantity (1430ms)
    ✓ can delete an issued refund (178ms)
```

Not sure if it's worth merging in master or re-running the tests (or even both) just to see if it's a hiccup in the pipeline.

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-edit.test.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        
+        it('can update order details', async () => {
+			//Open order we created
+			await merchant.goToOrder(orderId);
+
+			// Make sure we're still on the order details page
+			await expect(page.title()).resolves.toMatch('Edit order');
+
+			// Update order details
+			await expect(page).toFill('input[name=order_date]', '2018-12-14');
```
[the remark this answers is not in the document]
Comment (tammullen) [reply tests/e2e/core-tests/specs/merchant/wp-admin-order-edit.test.js:68]: Yes good point! Added a check.

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-edit.test.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+				// Wait for auto save
```
[the remark this answers is not in the document]
Comment (tammullen) [reply tests/e2e/core-tests/specs/merchant/wp-admin-order-edit.test.js]: Fixed, thanks for catching.

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-edit.test.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			await page.waitFor( 2000 );
+
+			// Create the order
```
[the remark this answers is not in the document]
Comment (tammullen) [reply tests/e2e/core-tests/specs/merchant/wp-admin-order-edit.test.js]: Agreed. Updated.

**Comment by tammullen** [comment]
> Not sure if it's worth merging in master or re-running the tests (or even both)

Did both, seems ok now.

**Comment by rrennick** [comment]
@tammullen This looks great to me. Leaving the approval for @zhongruige .

**Review by zhongruige — APPROVED** [review]
Changes look great, thanks for adding in this test @tammullen!

Runs great locally as well:

```
  WooCommerce Orders > Edit order
    ✓ can view single order (2215ms)
    ✓ can update order status (5098ms)
    ✓ can update order details (3976ms)
```

**Pull request merged**